src/dmd/dinterpret.d

```diff
@@ -797,6 +813,26 @@ public:
         Expression e = interpret(pue, s.exp, istate, CTFEGoal.Nothing);
         if (exceptionOrCant(e))
             return;
+
+        // destroy temporaries
+        // https://issues.dlang.org/show_bug.cgi?id=22536
+        auto dtors = ctfeGlobals.dtorsForTemporaries.copy();
+        ctfeGlobals.dtorsForTemporaries.setDim(0);
```

Comment:
I just noticed that setting the dim to 0 does not actually free any memory, it simply sets the length to 0. Since `struct Array` uses a dynamic array underneath, it should also set the length of it. Currently, the memory is not freed until the actual Expression pointers (in this case) are overwritten by later element additions.